app/models/manageiq/providers/vmware/infra_manager/refresh_parser/filter.rb

```diff
@@ -226,6 +234,36 @@ def rp_metadata_inv_by_vm_inv(vm_inv)
       rp_inv
     end
 
+    def storage_profile_inv_by_vm_inv(vm_inv)
+      storage_profile_inv           = {}
+      storage_profile_datastore_inv = {}
+      storage_profile_entity_inv    = {}
+
+      storage_profiles    = @vc_data[:storage_profile] || {}
+      matching_hubs       = @vc_data[:storage_profile_datastore] || {}
+      associated_entities = @vc_data[:storage_profile_entity] || {}
+
+      vm_refs = vm_inv.keys
+
+      associated_entities.each do |profile_id, profile_entities|
+        entities = profile_entities.to_a.select do |entity|
+          # An associated entity can either be a vm ref e.g.: 'vm-123'
+          # or a disk ref e.g.: 'vm-123:2000'
+          associated_vm = entity.key.split(':').first
```

Comment:
Since a disk will only show up here if its containing VM is associated with a profile, so we can simply check for VMs here. I.e. just check for ``` vm_refs.include?(associated_vm)```